src/main/java/codes/cookies/mod/features/farming/garden/visitors/VisitorDropProtection.java

```diff
@@ -0,0 +1,119 @@
+package codes.cookies.mod.features.farming.garden.visitors;
+
+import codes.cookies.mod.config.ConfigManager;
+import codes.cookies.mod.events.InventoryEvents;
+import codes.cookies.mod.events.api.InventoryContentUpdateEvent;
+import codes.cookies.mod.utils.cookies.CookiesUtils;
+import codes.cookies.mod.utils.exceptions.ExceptionHandler;
+import codes.cookies.mod.utils.items.CookiesDataComponentTypes;
+import codes.cookies.mod.utils.skyblock.LocationUtils;
+import codes.cookies.mod.utils.skyblock.inventories.ItemBuilder;
+import com.google.common.util.concurrent.Runnables;
+import net.minecraft.client.gui.screen.ingame.HandledScreen;
+import net.minecraft.component.ComponentType;
+import net.minecraft.component.DataComponentTypes;
+import net.minecraft.component.type.CustomModelDataComponent;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.screen.slot.Slot;
+import net.minecraft.text.Text;
+
+import net.minecraft.util.Rarity;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VisitorDropProtection {
+
+	private VisitorDropProtection(HandledScreen<?> handledScreen) {
+		InventoryContentUpdateEvent.registerSlot(handledScreen.getScreenHandler(),
+				ExceptionHandler.wrap(this::updateSlots));
+	}
+
+	private void updateSlots(Slot slot) {
+		if(slot.id == 33) {
+			handleRejectButton(slot.getStack(), handleAcceptItem(slot.inventory.getStack(29)));
+		}
+	}
+
+	protected final ItemStack disabledItem = new ItemBuilder(Items.BARRIER).hideAdditionalTooltips().set(CookiesDataComponentTypes.ITEM_CLICK_RUNNABLE, Runnables.doNothing()).setName(Text.of("Refusing temporarily disabled due to rare drop")).set(DataComponentTypes.RARITY, Rarity.COMMON).build();
+
+	private void handleRejectButton(ItemStack rejectButton, boolean visitorItemLore) {
+		if(rejectButton.isEmpty() || !rejectButton.isOf(Items.RED_TERRACOTTA) || !visitorItemLore) {
```

Comment:
In terms of consistency, there should be a null-check here as well
```suggestion
		if(rejectButton == null || rejectButton.isEmpty() || !rejectButton.isOf(Items.RED_TERRACOTTA) || !visitorItemLore) {
```